sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -844,6 +848,35 @@ mod checked {
                 }
             }
 
+            // Before finishing, enforce restrictions on transfer and deletion for objects configured
+            // with authenticators.
+            for (id, original_owner) in authenticator_objects {
+                let authenticator = original_owner.authenticator().expect("verified before adding to `authenticator_objects` that these have authenticators");
```

Comment:
since you don't use `original_owner` again after this, you could just make authenticator_objects be `Vec<(ObjectID, Authenticator)>` and then you could remove this line

Reply:
That's what I originally had, but then thought forward to when we have more authenticators than just SingleOwner, you are going to want to have the original owner for comparison as in the example on line 873